website_blog_excerpt_img/models/blog_post.py

```diff
@@ -1,32 +1,25 @@
 # Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
 
-import json
 from odoo import models
 
 
 class BlogPost(models.Model):
     _inherit = "blog.post"
 
-    def main_image(self):
-        """Get blog's main image URL."""
-        Converter = self.env['ir.fields.converter']
-        html = self.content
-        # Get a dictionary of properties, avoiding possible malformed ones
-        try:
-            properties = json.loads(self.cover_properties)
-        except (TypeError, ValueError):
-            properties = dict()
-        # Prepend cover image to post content, if there is one
-        cover = properties.pop("background-image", "none")
-        if cover and cover != "none":
-            html = u"<div style={q}background-image:{}{q}/>{}".format(
-                cover,
-                html,
-                q='"' if '"' not in cover else "'",
-            )
-        # Return the first found image URL or None
-        try:
-            return next(Converter.imgs_from_html(html, 1))
-        except StopIteration:
-            return None
+    def _default_website_meta(self):
```

Comment:
We should do this on each write if no previous default meta-image, not only on creation, or this is called each time?

Reply:
This is called when obtaining the social share details. [This is the method used to provide defaults](https://github.com/odoo/odoo/blob/3ac23bef79e3e0f8179c0e9441d6a81acdc55d1f/addons/website/models/website.py#L865-L867) (which is actually overriden in blog posts to return the blog cover, if set).

So, we don't want to write an option, but provide a default value, while still letting the user to choose a different value if he wants.

In short, this is the place to do this. :wink:

Reply:
I think you are not understanding me. Imagine this situation:

- You create a blog page with no image.
- Save.
- Then you edit again and add an image.

My question is: in this case, the later added image is going to be auto-selected?

Reply:
Yes.